### PR TITLE
add OSX support

### PR DESCRIPTION
**Pull request by fafhrd91**
this pull request adds OS X support and also adds support for iOS extensions.

### CODE REVIEW

**Comment by allanca** [comment]
Awesome, thank you! I'll check it out.

**Comment by allanca** [comment]
Looks great.

One addition: add a condition to the constants so the Platform constant matches the project target:

```
#if TARGET_OS_IPHONE
NSString *const kAMPPlatform = @"iOS";
#else
NSString *const kAMPPlatform = @"OS X";
#endif
```

Also, I'd want to get the tests building and passing against an OS X target as well so we have an easy way to make sure we're maintaining both targets. I may take a shot at this as part of deeper investigation anyway.

File: Amplitude/AMPDeviceInfo.m [hunk 1]
```diff
@@ -180,6 +207,75 @@ + (NSString*)getVendorID:(int) maxAttempts
     } else {
         return identifier;
     }
+#else
```
Comment (allanca) [inline Amplitude/AMPDeviceInfo.m]: Will this code ever run on OS X? Looks like the call to getVendorID:maxAttempts in getVendorId is compiled out.

**Comment by fafhrd91** [comment]
i'll update PR later today.

also i think `NSApplicationDidResignActiveNotification` notification does not properly work for OS X

**Comment by allanca** [comment]
Cool.

I've put my work on getting tests running here https://github.com/amplitude/Amplitude-iOS/tree/osx-cross-compile. I'm still working on a couple of cases that aren't passing.

Also, added one change to the automatic session starting for OS X.

**Comment by bradjasper** [comment]
Hello, I'm very interested in adding Amplitude to my Mac app. Is official support coming? I notice the current branch is a bit behind master. Are there any major issues to getting this implemented? I'd be happy to give it a shot if it needs some more work.

**Comment by djih** [comment]
Hi Brad,

Sorry for the slow response. Unfortunately we don't plan to add support for OS X in the immediate future. This repository is for iOS only, and we want to keep it as lightweight as possible. But definitely feel free to fork our repo and make your changes.

**Comment by bradjasper** [comment]
That's unfortunate as I'm sure many Mac apps would love to try Amplitude. I don't think maintaining a fork is sustainable if there's no plans to eventually support it. I hope you guys reconsider Mac support in the future—Amplitude looks great!

**Pull request closed without merging**